include/hip/hcc_detail/functional_grid_launch.hpp

```diff
@@ -176,4 +179,15 @@ void hipLaunchKernelGGL(F kernel, const dim3& numBlocks, const dim3& dimBlocks,
                                      numBlocks, dimBlocks, sharedMemBytes,
                                      stream, &config[0]);
 }
+
+inline
+__attribute__((visibility("hidden")))
+hipError_t hipExtLaunchMultiKernelMultiDevice(hipLaunchParams* launchParamsList,
+                                              int  numDevices, unsigned int  flags) {
+    hip_impl::hip_init();
+    auto& ps = hip_impl::get_program_state();
```

Comment:
Isn't  hip_impl::hip_init() required whether or not stub library is used?

Reply:
hip_impl::hip_init() should be called at the beginning of every HIP API regardless of using stub library or not. In the previous implementation of  the hipExtLaunchMultiKernelMultiDevice API the HIP_INIT_API() macro was doing the initialization by calling hip_impl::hip_init().

Reply:
Should similar changes be applied to functions like `hipLaunchCooperativeKernel()`?

Reply:
@srinivamd yes similar change would be required for hipLaunchCooperativeKernel and hipLaunchCooperativeKernelMultiDevice APIs which will be done in a separate PR later as these APIs are not ready yet and they are work in progress.